flow/project.py

```diff
@@ -1194,17 +1172,16 @@ def _get_submission_directives(self, default_directives, jobs):
         No checks are done to mitigate inappropriate aggregation of operations.
         This can lead to poor utilization of computing resources.
         """
-        directives = dict(ngpu=0, nranks=0, omp_num_threads=0, np=0)
-
-        for name in self.operations:
+        env = jobs[0]._project._environment
+        op_names = list(self.operations)
```

Comment:
Makes the structure of `self.operations` more explicit.
```suggestion
        op_names = list(self.operations.keys())
```